base_search_fuzzy/README.rst

```diff
@@ -0,0 +1,99 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=========================
+PostgreSQL Trigram Search
+=========================
+
+This addon provides the ability to create GIN or GiST indexes of char and text
+fields and also to use the search operator `%` in search domains. Currently
+this module doesn't change the backend search or something else. It provides
+only the possibilty to do a fuzzy search for external addons.
+
+
+Installation
+============
+
+First you need to have the ``pg_trgm`` extension available. In debian based
+distribution you have to install the `postgresql-contrib` module. Then you have
+to either install the ``pg_trgm`` extension to your database or you have to give
+your postgresql user the superadmin right (this allows the odoo module to
+install the extension to the database).
+
+
+Configuration
+=============
+
+If you installed the odoo module you can define ``GIN`` and ``GiST`` indexes for
+`char` and `text` via `Settings -> Database Structure -> Trigram Index`. The
+index name will automatically created for new entries.
+
+
+Usage
+=====
+
+For example you can create an index for the `name` field of `res.partner`. Then
+in a search you can use
+
+``self.env['res.partner'].search([('name', '%', 'Jon Miller)])``
+
+In this Example it can find existing names like `John Miller` or `John Mill`.
+
+Which strings can be found depends on the limit which is set (default: 0.3).
+Currently you have to set the limit by executing SQL:
+
+``self.env.cr.execute("SELECT set_limit(0.2);")``
+
+Also you can use the ``similarity(column, 'text')`` function in the ``order``
```

Comment:
Another interesting feature is the use of ``similarity(column, 'text')`` function in the ``order``
parameter to order by similarity. This module just contains a basic implementation which
doesn't perform validations and has to start with this function. For example
you can define the function as followed: